docs/app/pages/components/sections/keyboard/hotkeys/hotkeys.component.ts

```diff
@@ -0,0 +1,123 @@
+import { ColumnSortingComponent } from './../../../../../../../src/components/column-sorting/column-sorting.component';
+import { Component, Inject, ElementRef } from '@angular/core';
+import { DocumentationSectionComponent } from '../../../../../decorators/documentation-section-component';
+import { BaseDocumentationSection } from '../../../../../components/base-documentation-section/base-documentation-section';
+import { IPlunk } from '../../../../../interfaces/IPlunk';
+import { IPlunkProvider } from './../../../../../interfaces/IPlunkProvider';
+
+@Component({
+    selector: 'uxd-components-hotkeys',
+    templateUrl: './hotkeys.component.html',
+    styleUrls: ['./hotkeys.component.less'],
+    host: {'(window:keydown)': 'hotkeys($event)'}
+})
+@DocumentationSectionComponent('ComponentsHotkeysComponent')
+export class ComponentsHotkeysComponent extends BaseDocumentationSection implements IPlunkProvider {
+
+    nativeElement: HTMLElement;
+    qButtons: HTMLElement;
+    qSelected: number = 0;
+    qText: string = 'None';
+    wButtons: HTMLElement;
+    wSelected: number = 0;
+    wText: string = 'None';
+
+    constructor( @Inject(ElementRef) private element: ElementRef) {
+
+        super(
+            null, // require.context('!!prismjs-loader?lang=html!./snippets/', false, /\.html$/),
+            null, // require.context('!!prismjs-loader?lang=css!./snippets/', false, /\.css$/),
+            null, // require.context('!!prismjs-loader?lang=javascript!./snippets/', false, /\.js$/),
+            null, // require.context('!!prismjs-loader?lang=typescript!./snippets/', false, /\.ts$/),
+            require.context('./snippets/', false, /\.(html|less|js|ts)$/)
+        );
+
+        this.nativeElement = <HTMLElement> this.element.nativeElement;
+    }
+
+    hotkeys (event: KeyboardEvent) {
```

Comment:
We can simplify this and the view quite a lot:

**hotkey.component.ts**
```typescript
import { Component } from '@angular/core';
import { DocumentationSectionComponent } from '../../../../../decorators/documentation-section-component';
import { BaseDocumentationSection } from '../../../../../components/base-documentation-section/base-documentation-section';
import { IPlunk } from '../../../../../interfaces/IPlunk';
import { IPlunkProvider } from './../../../../../interfaces/IPlunkProvider';

@Component({
    selector: 'uxd-components-hotkeys',
    templateUrl: './hotkeys.component.html',
    styleUrls: ['./hotkeys.component.less'],
    host: {
        '(window:keydown.q)': 'focusNextQ()',
        '(window:keydown.w)': 'focusNextW()'
    }
})
@DocumentationSectionComponent('ComponentsHotkeysComponent')
export class ComponentsHotkeysComponent extends BaseDocumentationSection implements IPlunkProvider {

    qText: string = 'None';
    wText: string = 'None';
    qFocused: number = null;
    wFocused: number = null;

    plunk: IPlunk = {
        files: {
            'app.component.ts': require('./snippets/app.ts'),
            'app.component.html': require('./snippets/app.html'),
            'app.component.less': require('./snippets/app.less')
        }
    };

    constructor() {
        super(
            null, // require.context('!!prismjs-loader?lang=html!./snippets/', false, /\.html$/),
            null, // require.context('!!prismjs-loader?lang=css!./snippets/', false, /\.css$/),
            null, // require.context('!!prismjs-loader?lang=javascript!./snippets/', false, /\.js$/),
            null, // require.context('!!prismjs-loader?lang=typescript!./snippets/', false, /\.ts$/),
            require.context('./snippets/', false, /\.(html|less|js|ts)$www/)
        );
    }

    focusNextQ() {
        this.qFocused = this.qFocused === null || this.qFocused === 3 ? 0 : this.qFocused + 1;
    }

    focusNextW() {
        this.wFocused = this.wFocused === null || this.wFocused === 3 ? 0 : this.wFocused + 1;
    }

}
```

**hotkeys.component.html**
```html
<div class="hotkey-demo-box aspects-focus-outline" tabindex="0" [focusIf]="qFocused === 0">
    <div class="hotkey-demo-hint">
        <div class="hotkey-group-hint">q</div>
    </div>
    <div class="hotkey-demo-button-layout">
        <button class="btn button button-secondary" (keydown.space)="qText = 'Active'" (keydown.enter)="qText = 'Active'" (click)="qText = 'Active'" [focusIf]="qFocused === 1">
            <i class="hpe-icon hpe-active"></i>
        </button>
        <button class="btn button button-secondary" (keydown.space)="qText = 'Inactive'" (keydown.enter)="qText = 'Inactive'" (click)="qText = 'Inactive'" [focusIf]="qFocused === 2">
            <i class="hpe-icon hpe-inactive"></i>
        </button>
        <button class="btn button button-secondary" (keydown.space)="qText = 'Lock'" (keydown.enter)="qText = 'Lock'" (click)="qText = 'Lock'" [focusIf]="qFocused === 3">
            <i class="hpe-icon hpe-lock"></i>
        </button>
    </div>
    <p class="hotkey-demo-text">Selected: <span>{{ qText }}</span></p>
</div>

<div class="hotkey-demo-box aspects-focus-outline m-l-lg" tabindex="0" [focusIf]="wFocused === 0">
    <div class="hotkey-demo-hint">
        <div class="hotkey-group-hint">w</div>
    </div>
    <div class="hotkey-demo-button-layout">
        <button class="btn button button-secondary" (focus)="wText = 'Decelerate'" [focusIf]="wFocused === 1">
            <i class="hpe-icon hpe-rewind"></i>
        </button>
        <button class="btn button button-secondary" (focus)="wText = 'Drive'" [focusIf]="wFocused === 2">
            <i class="hpe-icon hpe-play"></i>
        </button>
        <button class="btn button button-secondary" (focus)="wText = 'Accelerate'" [focusIf]="wFocused === 3">
            <i class="hpe-icon hpe-fast-forward"></i>
        </button>
    </div>
    <p class="hotkey-demo-text">Selected: <span>{{ wText }}</span></p>
</div>
```

**New Directive - focusIf**

Create a new directive called focusIf that will allow us to conditionally focus an element. Make this part of UX Aspects library and document under utilites:

```typescript
import { Directive, Input, OnChanges, SimpleChanges, ElementRef } from '@angular/core';

@Directive({ selector: '[focusIf]' })
export class FocusIfDirective implements OnChanges {

    @Input() focusIf: boolean = false;

    constructor(private elementRef: ElementRef) { }

    ngOnChanges(changes: SimpleChanges): void {
        if (changes.focusIf && changes.focusIf.previousValue === false && changes.focusIf.currentValue === true) {
            this.elementRef.nativeElement.focus();
        }
    }
}
```